alacritty/src/renderer/solidrect.rs

```diff
@@ -0,0 +1,200 @@
+            );
+            gl::EnableVertexAttribArray(1);
+
+            gl::BindVertexArray(0);
+            gl::BindBuffer(gl::ARRAY_BUFFER, 0);
+        }
+
+        Self { vao, vbo, ebo, indices: Vec::new(), vertices: Vec::new(), uploaded_indices: 0 }
+    }
+
+    pub fn draw(
+        &mut self,
+        program: &RectShaderProgram,
+        size_info: &SizeInfo,
+        rects: Vec<RenderRect>,
+    ) {
+        if rects.is_empty() {
+            return;
+        }
+
+        // Prepare common state
+        unsafe {
+            // Remove padding from viewport.
+            gl::Viewport(0, 0, size_info.width() as i32, size_info.height() as i32);
+
+            gl::Enable(gl::BLEND);
+            gl::BlendFuncSeparate(gl::SRC_ALPHA, gl::ONE_MINUS_SRC_ALPHA, gl::SRC_ALPHA, gl::ONE);
+
+            // Setup bindings. VAO will set up attribs and EBO, but not VBO.
+            gl::BindVertexArray(self.vao);
+            gl::BindBuffer(gl::ARRAY_BUFFER, self.vbo);
+
+            gl::UseProgram(program.id);
+        }
+
+        let center_x = size_info.width() / 2.;
+        let center_y = size_info.height() / 2.;
+
+        for rect in &rects {
+            self.append_rect(center_x, center_y, rect);
+        }
+
+        self.draw_accumulated();
+    }
+
+    fn append_rect(&mut self, center_x: f32, center_y: f32, rect: &RenderRect) {
+        if rect.alpha <= 0. {
+            return;
+        }
+
+        assert!(self.vertices.len() <= MAX_U16_INDICES - 4);
+
+        // Calculate rectangle position.
+        let x = (rect.x - center_x) / center_x;
+        let y = -(rect.y - center_y) / center_y;
+        let width = rect.width / center_x;
+        let height = rect.height / center_y;
+        let color = Rgba {
+            r: rect.color.r,
+            g: rect.color.g,
+            b: rect.color.b,
+            a: (rect.alpha * 255.) as u8,
+        };
+
+        self.vertices.extend_from_slice(&[
+            Vertex { x, y, color },
+            Vertex { x, y: y - height, color },
+            Vertex { x: x + width, y, color },
+            Vertex { x: x + width, y: y - height, color },
+        ]);
+
+        if self.vertices.len() == MAX_U16_INDICES {
+            self.draw_accumulated();
+        }
+    }
+
+    fn draw_accumulated(&mut self) {
+        if self.vertices.is_empty() {
+            return;
+        }
+
+        // Generate new indices in index buffer on-demand
+        assert!(self.indices.len() % 6 == 0);
+        let generated_quads = (self.indices.len() / 6) as u16;
+        let need_quads = (self.vertices.len() / 4) as u16;
+        for index in generated_quads..need_quads {
+            let index = index * 4;
+            self.indices.extend_from_slice(&[
+                index,
+                index + 1,
+                index + 2,
+                index + 2,
+                index + 3,
+                index + 1,
+            ]);
+        }
+
+        // Upload accumulated buffers
```

Comment:
```suggestion
        // Upload accumulated buffers.
```

Generally comments should always end with a period. This applies to all comments in this PR.

Reply:
Fixed